lib/cocoapods/target/build_settings.rb

```diff
@@ -0,0 +1,1064 @@
+        flags
+      end
+
+      #-------------------------------------------------------------------------#
+
+      # @!group Linking
+
+      # @return [Boolean]
+      define_build_settings_method :requires_objc_linker_flag? do
+        false
+      end
+
+      # @return [Boolean]
+      define_build_settings_method :requires_fobjc_arc? do
+        false
+      end
+
+      # @return [Array<String>]
+      #   the `LD_RUNPATH_SEARCH_PATHS` needed for dynamically linking the {#target}
+      #
+      # @param [Boolean] requires_host_target
+      #
+      # @param [Boolean] test_bundle
+      #
+      def _ld_runpath_search_paths(requires_host_target: false, test_bundle: false)
+        if target.platform.symbolic_name == :osx
+          ["'@executable_path/../Frameworks'",
+           test_bundle ? "'@loader_path/../Frameworks'" : "'@loader_path/Frameworks'"]
+        else
+          paths = [
+            "'@executable_path/Frameworks'",
+            "'@loader_path/Frameworks'",
+          ]
+          paths << "'@executable_path/../../Frameworks'" if requires_host_target
+          paths
+        end
+      end
+      private :_ld_runpath_search_paths
+
+      # @return [Array<String>]
+      define_build_settings_method :other_ldflags, :build_setting => true, :memoized => true do
+        ld_flags = []
+        ld_flags << '-ObjC' if requires_objc_linker_flag?
+        if requires_fobjc_arc?
+          ld_flags << '-fobjc-arc'
+        end
+        libraries.each { |l| ld_flags << %(-l"#{l}") }
+        frameworks.each { |f| ld_flags << '-framework' << %("#{f}") }
+        weak_frameworks.each { |f| ld_flags << '-weak_framework' << %("#{f}") }
+        ld_flags
+      end
+
+      #-------------------------------------------------------------------------#
+
+      # @!group Private Methods
+
+      # Clears all memoized settings.
+      # @!visibility private
+      # @return [Void]
+      def __clear__
+        @__memoized = nil
+      end
+
+      private
+
+      # @return [Hash<String => String|Array<String>>]
+      def to_h
+        hash = {}
+        self.class.build_settings_names.sort.each do |setting|
+          hash[setting] = public_send(setting.downcase)
+        end
+        hash
+      end
+
+      # @return [Hash<String => String>]
+      def add_inherited_to_plural(hash)
+        Hash[hash.map do |key, value|
+          next [key, '$(inherited)'] if value.nil?
+          if PLURAL_SETTINGS.include?(key)
+            raise ArgumentError, "#{key} is a plural setting, cannot have #{value.inspect} as its value" unless value.is_a? Array
+
+            value = "$(inherited) #{quote_array(value)}"
+          else
+            raise ArgumentError, "#{key} is not a plural setting, cannot have #{value.inspect} as its value" unless value.is_a? String
+          end
+
+          [key, value]
+        end]
+      end
+
+      # @return [Array<String>]
+      #
+      # @param  [Array<String>] array
```

Comment:
these docs are not helpful can be removed or expanded.

Reply:
they give types for the parameters / return value